From f1652c724715af007c99051cfaf609157a69894f Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Tue, 5 Apr 2022 13:56:33 +0800
Subject: [PATCH] cheese: limitation the max resolution to 3840x2160

Signed-off-by: Haihua Hu <jared.hu@nxp.com>
---
 libcheese/cheese-camera-device.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/libcheese/cheese-camera-device.c b/libcheese/cheese-camera-device.c
index c916258e..a7d6dd50 100644
--- a/libcheese/cheese-camera-device.c
+++ b/libcheese/cheese-camera-device.c
@@ -475,10 +475,10 @@ cheese_camera_device_update_format_table (CheeseCameraDevice *device)
       if (min_height < 120)
         min_height = 120;
 
-      if (max_width > 5120)
-        max_width = 5120;
-      if (max_height > 3840)
-        max_height = 3840;
+      if (max_width > 3840)
+        max_width = 3840;
+      if (max_height > 2160)
+        max_height = 2160;
 
       cur_width  = min_width;
       cur_height = min_height;
-- 
2.17.1

