From 136aa53d70ab3504f996d097316c22d81bc86103 Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Thu, 4 Mar 2021 10:49:47 +0800
Subject: [PATCH 2/2] don't build help folder to avoid build break

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index a4775d44..90ee2935 100644
--- a/meson.build
+++ b/meson.build
@@ -183,7 +183,7 @@ if get_option('tests')
 endif
 
 subdir('po')
-subdir('help')
+#subdir('help')
 
 enable_gtk_doc = get_option('gtk_doc')
 enable_man = get_option('man')
-- 
2.17.1

