From 36dbd3cdbd9baf10df1c9a4d28ac182679eb16a4 Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Thu, 23 Sep 2021 13:30:14 +0800
Subject: [PATCH] add imxvideoconvert_g2d into playsink

---
 gst/playback/gstplaysink.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/gst/playback/gstplaysink.c b/gst/playback/gstplaysink.c
index 2c080bc59..843b8a9e4 100755
--- a/gst/playback/gstplaysink.c
+++ b/gst/playback/gstplaysink.c
@@ -692,6 +692,12 @@ gst_play_sink_init (GstPlaySink * playsink)
   gst_bin_add (GST_BIN_CAST (playsink),
       GST_ELEMENT_CAST (playsink->stream_synchronizer));
 
+  playsink->video_filter =
+      gst_element_factory_make ("imxvideoconvert_g2d", "g2dconvert");
+  if (!playsink->video_filter) {
+    GST_WARNING ("imxvideoconvert_g2d is not installed");
+  }
+
   g_rec_mutex_init (&playsink->lock);
   GST_OBJECT_FLAG_SET (playsink, GST_ELEMENT_FLAG_SINK);
   gst_bin_set_suppressed_flags (GST_BIN (playsink),
-- 
2.17.1

