From 654fc7356afb40700887ff3e91d3732dec341fcc Mon Sep 17 00:00:00 2001
From: nxa15299 <Heinz.Wrobel@nxp.com>
Date: Tue, 23 Jan 2018 08:33:06 +0100
Subject: [PATCH] pseudo: Another missing EAR3 patch

Signed-off-by: nxa15299 <Heinz.Wrobel@nxp.com>

---
 pseudo_util.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/pseudo_util.c b/pseudo_util.c
index d1c79c4..9e6e16c 100644
--- a/pseudo_util.c
+++ b/pseudo_util.c
@@ -807,7 +807,9 @@ pseudo_fix_path(const char *base, const char *path, size_t rootlen, size_t basel
 	char *current;
 	char *effective_root;
 	int trailing_slash = 0;
-	
+
+	if (!readlinkp)
+		readlinkp = readlink;
 	if (!path) {
 		pseudo_diag("can't fix empty path.\n");
 		return 0;
