From 78baf18033f5578373491185d125f1d33e3dda57 Mon Sep 17 00:00:00 2001
From: nxa15299 <Heinz.Wrobel@nxp.com>
Date: Sat, 27 Jan 2018 11:18:46 +0100
Subject: [PATCH] pseudo: realpath wrapper did not chroot translation properly

Signed-off-by: nxa15299 <Heinz.Wrobel@nxp.com>
---
 ports/unix/guts/realpath.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/ports/unix/guts/realpath.c b/ports/unix/guts/realpath.c
index a59808d..61a3c0d 100644
--- a/ports/unix/guts/realpath.c
+++ b/ports/unix/guts/realpath.c
@@ -12,7 +12,10 @@
 		errno = ENAMETOOLONG;
 		return NULL;
 	}
-	if ((len = strlen(rname)) >= pseudo_sys_path_max()) {
+	len = strlen(rname);
+	len = pseudo_dechroot(rname, len);
+
+	if (len >= pseudo_sys_path_max()) {
 		errno = ENAMETOOLONG;
 		return NULL;
 	}
-- 
2.15.1

