From 240409906dc030a7a5160cef31560aa638b3532f Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Fri, 21 Dec 2018 17:56:04 +0200
Subject: [PATCH] pseudo: Cleaned check of symlink flag handling

Signed-off-by: nxa15299 <Heinz.Wrobel@nxp.com>
Signed-off-by: Costea Ciprian <ciprianmarian.costea@nxp.com>

---
 pseudo_util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pseudo_util.c b/pseudo_util.c
index 51782a7..d1c79c4 100644
--- a/pseudo_util.c
+++ b/pseudo_util.c
@@ -680,7 +680,7 @@ pseudo_append_element(char *newpath, char *root, size_t allocated, char **pcurre
 	 * were appending to. we don't want to actually try to do this when
 	 * we're appending names to a regular file.
 	 */
-	if (!leave_this && is_dir) {
+	if ( (!(leave_this & AT_SYMLINK_NOFOLLOW)) && is_dir) {
 		int is_link = S_ISLNK(buf->st_mode);
 		if (link_recursion >= PSEUDO_MAX_LINK_RECURSION && is_link) {
 			pseudo_debug(PDBGF_PATH, "link recursion too deep, not expanding path '%s'.\n", newpath);
