From 925408d5acb79deb28e1f6d867436306018cec66 Mon Sep 17 00:00:00 2001
From: Zelan Zou <zelan.zou@nxp.com>
Date: Tue, 6 Jun 2023 15:30:33 +0800
Subject: [PATCH] attr/attributes.h is providing the ENOATTR definition.

Signed-off-by: Zelan Zou <zelan.zou@nxp.com>
---
 ports/linux/xattr/portdefs.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ports/linux/xattr/portdefs.h b/ports/linux/xattr/portdefs.h
index beab7d0..89aa885 100644
--- a/ports/linux/xattr/portdefs.h
+++ b/ports/linux/xattr/portdefs.h
@@ -3,6 +3,7 @@
  *
  */
 #include <sys/xattr.h>
+#include <attr/attributes.h>
 #include <stdint.h>
 
 #ifndef ENOATTR
-- 
2.25.1

