From 413cc734d37876361137b7235a2636b1f2a41e55 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Tue, 8 Nov 2022 11:29:03 +0100
Subject: [PATCH 02/20] pseudo: The stat[v]fs() functions were not wrapped
 properly

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
---
 ports/linux/wrapfuncs.in | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/ports/linux/wrapfuncs.in b/ports/linux/wrapfuncs.in
index 97b16c2..241cf52 100644
--- a/ports/linux/wrapfuncs.in
+++ b/ports/linux/wrapfuncs.in
@@ -22,12 +22,20 @@ int openat64(int dirfd, const char *path, int flags, ...{mode_t mode}); /* flags
 int __openat64_2(int dirfd, const char *path, int flags); /* flags=flags&O_NOFOLLOW, noignore_path=1 */
 int creat64(const char *path, mode_t mode);
 int stat(const char *path, struct stat *buf); /* real_func=pseudo_stat */
+int statfs(const char *path, struct statfs *buf);
+int statvfs(const char *path, struct statvfs *buf);
 int lstat(const char *path, struct stat *buf); /* real_func=pseudo_lstat, flags=AT_SYMLINK_NOFOLLOW */
 int fstat(int fd, struct stat *buf); /* real_func=pseudo_fstat */
+int fstatfs(int fd, struct statfs *buf);
+int fstatvfs(int fd, struct statvfs *buf);
 int fstatat(int dirfd, const char *path, struct stat *buf, int flags);
 int stat64(const char *path, struct stat64 *buf); /* real_func=pseudo_stat64 */
+int statfs64(const char *path, struct statfs64 *buf);
+int statvfs64(const char *path, struct statvfs64 *buf);
 int lstat64(const char *path, struct stat64 *buf); /* real_func=pseudo_lstat64, flags=AT_SYMLINK_NOFOLLOW */
 int fstat64(int fd, struct stat64 *buf); /* real_func=pseudo_fstat64 */
+int fstatfs64(int fd, struct statfs64 *buf);
+int fstatvfs64(int fd, struct statvfs64 *buf);
 int fstatat64(int dirfd, const char *path, struct stat64 *buf, int flags);
 int __xstat64(int ver, const char *path, struct stat64 *buf);
 int __lxstat64(int ver, const char *path, struct stat64 *buf); /* flags=AT_SYMLINK_NOFOLLOW */
-- 
2.34.1

