From fc20763b9594e67399b0d05dca358b7f55ca514f Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Tue, 8 Nov 2022 11:40:15 +0100
Subject: [PATCH 07/20] pseudo: path search was not shell compliant

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
---
 pseudo_client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pseudo_client.c b/pseudo_client.c
index f75ce1c..3165c4e 100644
--- a/pseudo_client.c
+++ b/pseudo_client.c
@@ -2270,7 +2270,7 @@ pseudo_exec_path(const char *filename, int search_path) {
 		return candidate;
 	}
 
-	if (!search_path || !path_segs) {
+	if (!search_path || !path_segs || strchr(filename, '/')) {
 		candidate = pseudo_fix_path(pseudo_cwd, filename, 0, pseudo_cwd_len, NULL, 0);
 		/* executable or not, it's the only thing we can try */
 		pseudo_magic();
-- 
2.34.1

