From fe2006f4d190bbb4c3804dcadc78f56c18bb3a7c Mon Sep 17 00:00:00 2001
From: r39252 <Heinz.Wrobel@freescale.com>
Date: Thu, 19 Oct 2017 07:16:50 +0200
Subject: [PATCH] pseudo: path search was not shell compliant

---
 pseudo_client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pseudo_client.c b/pseudo_client.c
index f75ce1c..3165c4e 100644
--- a/pseudo_client.c
+++ b/pseudo_client.c
@@ -2270,7 +2270,7 @@ pseudo_exec_path(const char *filename, int search_path) {
 		return candidate;
 	}
 
-	if (!search_path || !path_segs) {
+	if (!search_path || !path_segs || strchr(filename, '/')) {
 		candidate = pseudo_fix_path(pseudo_cwd, filename, 0, pseudo_cwd_len, NULL, 0);
 		/* executable or not, it's the only thing we can try */
 		pseudo_magic();
