From da0c3018c6c6c54a0dcd3a82df00b86a3b41ba51 Mon Sep 17 00:00:00 2001
From: r39252 <Heinz.Wrobel@freescale.com>
Date: Thu, 19 Oct 2017 07:19:40 +0200
Subject: [PATCH] pseudo: Oops. Forgot to add variable cleanup to support prior
 filename translation commit

---
 pseudo_client.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/pseudo_client.c b/pseudo_client.c
index 3165c4e..ccdbbc3 100644
--- a/pseudo_client.c
+++ b/pseudo_client.c
@@ -2231,8 +2231,9 @@ populate_path_segs(void) {
 
 const char *
 pseudo_exec_path(const char *filename, int search_path) {
+    char *s;
 	char *path = getenv("PATH");
-	char *candidate;
+	const char *candidate;
 	int i;
 	struct stat buf;
 	char aliasbuf[NAME_MAX];
@@ -2306,7 +2307,8 @@ pseudo_exec_path(const char *filename, int search_path) {
 		}
 	}
 	/* blind guess being as good as anything */
+	candidate = filename;
 	pseudo_magic();
-	return filename;
+	return candidate;
 }
 
