From 17c9d576b1fdc4b2d3c1a6c84c02ddd676d51de1 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Sat, 25 Jan 2020 15:03:42 +0100
Subject: [PATCH] qemu-user: strace didn't print faccessat() correctly

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
---
 linux-user/strace.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/linux-user/strace.c b/linux-user/strace.c
index 0edb759..1a9cc85 100644
--- a/linux-user/strace.c
+++ b/linux-user/strace.c
@@ -1413,7 +1413,8 @@ print_faccessat(const struct syscallname *name,
     print_at_dirfd(arg0, 0);
     print_string(arg1, 0);
     print_flags(access_flags, arg2, 0);
-    print_flags(at_file_flags, arg3, 1);
+    /* Kernel does not receive arg3! We pretend! */
+    gemu_log("?");
     print_syscall_epilogue(name);
 }
 #endif
-- 
2.17.1

