From 4e2ffec525ce9db1fad1383ad87726224325ae3c Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Thu, 23 Sep 2021 15:06:08 +0800
Subject: [PATCH] don't remove v4l2 plugin, it's the only decoder on 8qm

---
 gst/gstregistry.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gst/gstregistry.c b/gst/gstregistry.c
index 3ef4d8642..9195a6687 100644
--- a/gst/gstregistry.c
+++ b/gst/gstregistry.c
@@ -539,6 +539,11 @@ gst_registry_remove_plugin (GstRegistry * registry, GstPlugin * plugin)
   GST_DEBUG_OBJECT (registry, "removing plugin %p (%s)",
       plugin, gst_plugin_get_name (plugin));
 
+  if (g_strcmp0 ("video4linux2", gst_plugin_get_name (plugin)) == 0) {
+    GST_DEBUG_OBJECT (registry, "ignore request for remove v4l2 plugin");
+    return;
+  }
+
   GST_OBJECT_LOCK (registry);
   registry->priv->plugins = g_list_remove (registry->priv->plugins, plugin);
   --registry->priv->n_plugins;
-- 
2.17.1

